tests/object_file_system_test.php

```diff
@@ -206,16 +206,66 @@ public function test_delete_object_from_local_by_hash_if_not_local() {
 
     public function test_delete_object_from_local_by_hash_if_can_verify_external_object() {
         $file = $this->create_duplicated_file();
-        $externalpath = $this->get_external_path_from_hash($file->get_contenthash());
+        $contenthash = $file->get_contenthash();
+        $externalpath = $this->get_external_path_from_hash($contenthash);
         $localpath = $this->get_local_path_from_storedfile($file);
 
         $differentfilepath = __DIR__ . '/fixtures/test.txt';
         copy($differentfilepath, $externalpath);
+        // Get the grandparent path.
+        $grandparentpath = realpath(str_replace($contenthash, '', $localpath) . '..');
 
-        $location = $this->filesystem->delete_object_from_local_by_hash($file->get_contenthash());
+        $location = $this->filesystem->delete_object_from_local_by_hash($contenthash);
 
         $this->assertEquals(OBJECT_LOCATION_EXTERNAL, $location);
         $this->assertFalse(is_readable($localpath));
+        // If grandparent is empty it should be removed.
+        $this->assertFalse(is_readable($grandparentpath));
```

Comment:
would file_exists() which can be used for dirs, be better than is_readable?

Reply:
I used the same for consistency, there is not much difference in these cases.